Src/Base/AMReX_MultiFab.cpp

```diff
@@ -1187,29 +1187,63 @@ MultiFab::negate (const Box& region, int comp, int num_comp, int nghost)
     FabArray<FArrayBox>::mult(-1.,region,comp,num_comp,nghost);
 }
 
+void
+MultiFab::SumBoundary (const Periodicity& period)
+{
+    SumBoundary(0, n_comp, IntVect(0), period);
+}
+
+void
+MultiFab::SumBoundary (int scomp, int ncomp, const Periodicity& period)
+{
+    SumBoundary(scomp, ncomp, IntVect(0), period);
+}
+
 void
 MultiFab::SumBoundary (int scomp, int ncomp, IntVect const& nghost, const Periodicity& period)
 {
     BL_PROFILE("MultiFab::SumBoundary()");
 
-    if ( n_grow == IntVect::TheZeroVector() && boxArray().ixType().cellCentered()) return;
+    SumBoundary_nowait(scomp, ncomp, nghost, period);
+    SumBoundary_finish();
+}
 
-    MultiFab tmp(boxArray(), DistributionMap(), ncomp, n_grow, MFInfo(), Factory());
-    MultiFab::Copy(tmp, *this, scomp, 0, ncomp, n_grow);
-    this->setVal(0.0, scomp, ncomp, nghost);
-    this->ParallelCopy(tmp,0,scomp,ncomp,n_grow,nghost,period,FabArrayBase::ADD);
+void
+MultiFab::SumBoundary_nowait (const Periodicity& period)
+{
+    SumBoundary_nowait(0, n_comp, IntVect(0), period);
 }
 
 void
-MultiFab::SumBoundary (int scomp, int ncomp, const Periodicity& period)
+MultiFab::SumBoundary_nowait (int scomp, int ncomp, const Periodicity& period)
 {
-    SumBoundary(scomp, ncomp, IntVect(0), period);
+    SumBoundary_nowait(scomp, ncomp, IntVect(0), period);
 }
 
 void
-MultiFab::SumBoundary (const Periodicity& period)
+MultiFab::SumBoundary_nowait (int scomp, int ncomp, IntVect const& nghost, const Periodicity& period)
 {
-    SumBoundary(0, n_comp, IntVect(0), period);
+    BL_PROFILE("MultiFab::SumBoundary_nowait()");
+
+    if ( n_grow == IntVect::TheZeroVector() && boxArray().ixType().cellCentered()) return;
+
+    MultiFab* tmp = new MultiFab( boxArray(), DistributionMap(), ncomp, n_grow, MFInfo(), Factory() );
+    MultiFab::Copy(*tmp, *this, scomp, 0, ncomp, n_grow);
+    this->setVal(0.0, scomp, ncomp, nghost);
+    this->ParallelCopy_nowait(*tmp,0,scomp,ncomp,n_grow,nghost,period,FabArrayBase::ADD);
```

Comment:
There is a memory leak here.  Need to test if this->pcd is built.  If not delete tmp.